### PR TITLE
Parse followers audience from followers URI

### PR DESCRIPTION
**Pull request by InspectorCaracal**
Adds followersUri to user model and fetches from followers attribute in ActivityPub actor model.

This allows parsing the audience to use the actor's defined followers collection instead of the hardcoded path.

### CODE REVIEW

File: src/remote/activitypub/audience.ts [hunk 1]
```diff
@@ -88,6 +88,6 @@ function isPublic(id: string) {
 function isFollowers(id: string, actor: IRemoteUser) {
 	return [
 		`${actor.uri}/followers`,
-		// actor.followerUri,	// TODO
+		actor.followersUri,
```
Comment (mei23) [inline src/remote/activitypub/audience.ts]: If it's null, I don't want to use it.

File: src/remote/activitypub/models/person.ts [hunk 2]
```diff
@@ -330,6 +331,7 @@ export async function updatePerson(uri: string, resolver?: Resolver | null, hint
 		lastFetchedAt: new Date(),
 		inbox: person.inbox,
 		sharedInbox: person.sharedInbox || (person.endpoints ? person.endpoints.sharedInbox : undefined),
+		followersUri: person.followers,
```
Comment (mei23) [inline src/remote/activitypub/models/person.ts]: I think it's better to do the same as createPerson above.

Comment (InspectorCaracal) [reply src/remote/activitypub/models/person.ts]: good catch, thank you!

File: src/remote/activitypub/audience.ts (hunk 1 above)
Comment (InspectorCaracal) [reply src/remote/activitypub/audience.ts]: good point!

to fix this i have redone `isFollowers` to compare `id` to either `followersUri` or the default

an account should only have one followers collection, so the default should only be as a backup

File: src/remote/activitypub/audience.ts [hunk 3]
```diff
@@ -86,8 +86,7 @@ function isPublic(id: string) {
 }
 
 function isFollowers(id: string, actor: IRemoteUser) {
-	return [
-		`${actor.uri}/followers`,
-		// actor.followerUri,	// TODO
-	].includes(id);
+	return (
+		id == (actor.followersUri ? actor.followersUri : `${actor.uri}/followers`)
+	);
```
Comment (mei23) [inline src/remote/activitypub/audience.ts]: Please use `===`
And, maybe below is enough.
```suggestion
  return id === (actor.followersUri || `${actor.uri}/followers`);
```

Comment (InspectorCaracal) [reply src/remote/activitypub/audience.ts]: `||` will give a result of either True or False, not the URI, but I will use `===`

Comment (mei23) [reply src/remote/activitypub/audience.ts]: > `||` will give a result of either True or False

That's no problem
image

Comment (mei23) [reply src/remote/activitypub/audience.ts]: Perhaps its specification is only equality operators.
image

Comment (InspectorCaracal) [reply src/remote/activitypub/audience.ts]: oh! I learned something, thank you!

**Review by mei23 — APPROVED** [review]
カッコがもう1段不要だけどめんどくさいからいいわ

**Comment by syuilo** [comment]
Thank you guys🥰

**Pull request merged**